ops/testing.py

```diff
@@ -187,6 +194,23 @@ def _event_context(self, event_name: str):
         """
         return self._framework._event_context(event_name)
 
+    def emit_event(self, evt: BoundEvent, *args: Any, **kwargs: Any):
+        """Emit a bound event on the charm."""
+        if evt.emitter.framework is not self._framework:
+            # namely, not sure what might happen if it doesn't.
+            raise RuntimeError('event needs to be bound to an emitter '
+                               'sharing the same framework as this harness.')
+
+        if REINITIALIZE_CHARM_ON_EVENT:
```

Comment:
I'm a bit concerned about these being module values. It means that if you have any test that starts opting into this behavior, suddenly the rest of your test suite becomes a bit indeterminate.